freqtrade/freqai/data_kitchen.py

```diff
@@ -1173,10 +1217,13 @@ def use_strategy_to_populate_indicators(
                 informative=base_dataframes[tf],
                 set_generalized_indicators=sgi
             )
-            if pairs:
-                for i in pairs:
-                    if pair in i:
-                        continue  # dont repeat anything from whitelist
+
+        # ensure corr pairs are always last
+        for i in pairs:
+            if pair in i:
+                continue  # dont repeat anything from whitelist
+            for tf in tfs:
+                if pairs and do_corr_pairs:
                     dataframe = strategy.populate_any_indicators(
```

Comment:
I don't quite like variable naming here ... it's unclear what `i` is (is it a string? a list of strings? ... - someone not knowing freqAI insideout would have to look up the type in the config ... ) - maybe we can enhance the variables with typehints to help this?

Reply:
Sure, type hinting pairs, no problem. I thought you asked "it is unclear what `i` is"

Reply:
well that too `i` is kinda a bad name for anything :laughing: 
(it's actually a pair i think - though it could also be "coin" ... )

Reply:
This is resolved by typing the relevant lists and changing `i` to `corr_pair`.